--- complete.tcsh.orig	Sun Jul  7 15:37:20 2002
+++ complete.tcsh	Tue Jan 14 15:28:42 2003
@@ -639,10 +639,14 @@
     endif
     complete unsetenv	n/*/e/
 
-    if (-r /var/spool/mail) then
-      set _maildir = /var/spool/mail
+    if (-r /var/mail) then
+      set _maildir = /var/mail
     else
-      set _maildir = /usr/mail
+      if (-r /var/spool/mail) then
+        set _maildir = /var/spool/mail
+      else
+        set _maildir = /usr/mail
+      endif
     endif
     if (-r $HOME/.mailrc) then
         complete mail	c/-/"(e i f n s u v)"/ c/*@/\$hosts/ \
