--- tcsh-6.10.00/tc.who.c.orig	Thu Mar 22 08:48:11 2001
+++ tcsh-6.10.00/tc.who.c	Thu Mar 22 08:53:20 2001
@@ -241,22 +241,6 @@
     ncbs_posted = 0;
 #else /* !WINNT_NATIVE */
 
-    /*
-     * From: Michael Schroeder <mlschroe@immd4.informatik.uni-erlangen.de>
-     * Don't open utmp all the time, stat it first...
-     */
-    if (stat(_PATH_UTMP, &sta)) {
-	if (!force)
-	    xprintf(CGETS(26, 1,
-			  "cannot stat %s.  Please \"unset watch\".\n"),
-		    _PATH_UTMP);
-# ifdef BSDSIGS
-	(void) sigsetmask(omask);
-# else
-	(void) sigrelse(SIGINT);
-# endif
-	return;
-    }
     if (stlast == sta.st_mtime) {
 # ifdef BSDSIGS
 	(void) sigsetmask(omask);
