--- tcsh-6.09.00/sh.h.orig	Tue Mar 20 12:51:49 2001
+++ tcsh-6.09.00/sh.h	Tue Mar 20 12:56:36 2001
@@ -223,6 +223,8 @@
 #endif /* !_MINIX && !_VMS_POSIX && !WINNT */
 #include <sys/stat.h>
 
+#include <time.h>
+
 #if defined(BSDTIMES) || defined(BSDLIMIT)
 # include <sys/time.h>
 # if SYSVREL>3 && !defined(SCO) && !defined(sgi) && !defined(SNI) && !defined(sun) && !(defined(__alpha) && defined(__osf__)) && !defined(_SX)
