diff -urNpa tcsh-6.20.00.orig/sh.h tcsh-6.20.00/sh.h
--- tcsh-6.20.00.orig/sh.h	2018-01-12 17:40:23.000000000 +0000
+++ tcsh-6.20.00/sh.h	2018-01-12 17:42:56.464101352 +0000
@@ -261,7 +261,7 @@ typedef long tcsh_number_t;
 # include <sys/procstats.h>
 #endif /* _SEQUENT_ */
 #if (defined(POSIX) || SYSVREL > 0) && !defined(WINNT_NATIVE)
-# include <sys/times.h>
+# include <time.h>
 #endif /* (POSIX || SYSVREL > 0) && !WINNT_NATIVE */
 
 #ifdef NLS
