--- tcsh-6.13.00/nls/Makefile.orig	2004-02-21 21:34:26.000000000 +0100
+++ tcsh-6.13.00/nls/Makefile	2004-09-05 10:54:43.930185944 +0200
@@ -3,7 +3,7 @@
 		spanish ukrainian
 #GENCAT=	gencat --new
 #CHARSET=	charset
-GENCAT=		gencat
+GENCAT=		gencat --new
 
 catalogs:
 	@for i in ${SUBDIRS} ; \
--- tcsh-6.13.00/Makefile.in.orig	2003-03-12 20:14:50.000000000 +0100
+++ tcsh-6.13.00/Makefile.in	2004-09-05 10:55:21.588461016 +0200
@@ -136,8 +136,8 @@
 #DFLAGS=
 #DFLAGS=-D_PATH_TCSHELL='"${DESTBIN}/tcsh"'
 ## The following is set by autoconf.
-DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' @DFLAGS@ @CPPFLAGS@
-
+#DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' @DFLAGS@ @CPPFLAGS@
+DFLAGS = @DFLAGS@ @CPPFLAGS@
 
 ################################################################
 ## LDFLAGS.  Define something here if you need to
--- tcsh-6.08.01.orig/config_f.h
+++ tcsh-6.08.01/config_f.h
@@ -65,7 +65,7 @@
  *		if you don't have <nl_types.h>, you don't want
  *		to define this.
  */
-#undef NLS_CATALOGS
+#define NLS_CATALOGS
 
 /*
  * LOGINFIRST   Source ~/.login before ~/.cshrc
@@ -80,7 +80,7 @@
 /*
  * KAI          use "bye" command and rename "log" to "watchlog"
  */
-#undef KAI
+#define KAI
 
 /*
  * TESLA	drops DTR on logout. Historical note:
@@ -92,12 +92,12 @@
 /*
  * DOTLAST      put "." last in the default path, for security reasons
  */
-#define DOTLAST
+#undef DOTLAST
 
 /*
  * NODOT	Don't put "." in the default path, for security reasons
  */
-#undef NODOT
+#define NODOT
 
 /*
  * AUTOLOGOUT	tries to determine if it should set autologout depending
@@ -136,7 +136,7 @@
 /*
  * NEWGRP	Provide a newgrp builtin.
  */
-#undef NEWGRP
+#define NEWGRP
 
 /*
  * SYSMALLOC	Use the system provided version of malloc and friends.
--- tcsh-6.08.01.orig/tc.os.h
+++ tcsh-6.08.01/tc.os.h
@@ -529,7 +529,7 @@
 
 #if !defined(BSD4_4) && !defined(__linux__) && !defined(__hpux) && !defined(sgi)
 #ifndef NEEDgethostname
-extern int gethostname __P((char *, int));
+extern int gethostname __P((char *, size_t));
 #endif /* NEEDgethostname */
 #endif /* !BDS4_4 && !__linux__ && !__hpux && !sgi */
 
--- tcsh-6.08.01.orig/tcsh.man
+++ tcsh-6.08.01/tcsh.man
@@ -560,7 +560,7 @@
 changes to the next previous word etc., skipping identical matches
 much like \fIhistory-search-backward\fR does.
 .TP 8
-.B delete-char \fR(not bound)
+.B delete-char \fR(bound to `Del' if using the standard \fI/etc/csh.cshrc\fR)
 Deletes the character under the cursor.
 See also \fIdelete-char-or-list-or-eof\fR.
 .TP 8
