--- tcsh-6.13.00/sh.set.c.orig	2004-09-05 12:10:33.633525000 +0200
+++ tcsh-6.13.00/sh.set.c	2004-09-05 12:50:58.626870880 +0200
@@ -1255,7 +1255,7 @@
 
     for (i = 0; dspmt[i].n; i++) {
 	Char *estr;
-	if (t_pmatch(pcp, dspmt[i].n, &estr, 1) > 0) {
+	if (dspmt[i].n[0] && t_pmatch(pcp, dspmt[i].n, &estr, 1) > 0) {
 	    set(CHECK_MBYTEVAR, Strsave(dspmt[i].v), VAR_READWRITE);
 	    update_dspmbyte_vars();
 	    break;
