diff -urNpa tcsh-6.20.00.orig/sh.func.c tcsh-6.20.00/sh.func.c
--- tcsh-6.20.00.orig/sh.func.c	2018-01-12 17:44:47.000000000 +0000
+++ tcsh-6.20.00/sh.func.c	2018-01-12 17:47:14.904085948 +0000
@@ -2358,7 +2358,7 @@ setlim(struct limits *lp, int hard, RLIM
     if (setrlimit(lp->limconst, &rlim) < 0) {
        if ( ( lp->limconst == RLIMIT_LOCKS ) &&
 	    ( errno == EINVAL ) ) {
-	       xprintf(CGETS(15, 5, "cannot set it on the current kernel\n"));
+	       xprintf("%s", CGETS(15, 5, "cannot set it on the current kernel\n"));
 	       return(0);
        }
 # else /* BSDLIMIT */
