--- tcsh-6.13.00/config/linux~	2001-04-25 23:55:50.000000000 +0000
+++ tcsh-6.13.00/config/linux	2005-04-17 00:30:47.129774424 +0000
@@ -151,6 +151,11 @@
 /* have been moved to config_f.h */
 #include "config_f.h"
 
+#ifndef SYSMALLOC
+#define SYSMALLOC
+#endif
+
+
 #define NLS_CATALOGS
 
 #endif /* _h_config */
--- tcsh-6.17.00/config_f.h~	2011-09-01 20:39:09.000000000 +0200
+++ tcsh-6.17.00/config_f.h	2011-09-01 20:41:26.645696471 +0200
@@ -143,11 +143,7 @@
  *		This can be much slower and no memory statistics will be
  *		provided.
  */
-#if defined(__MACHTEN__) || defined(PURIFY) || defined(MALLOC_TRACE) || defined(_OSD_POSIX) || defined(__MVS__)
 # define SYSMALLOC
-#else
-# undef SYSMALLOC
-#endif
 
 /*
  * USE_ACCESS	Use access(2) rather than stat(2) when POSIX is defined.
